test/test-render-relationship.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2018  Eshan Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import {expect} from 'chai';
+import renderRelationship from '../src/server/helpers/render.js';
+
+
+const relationshipTests = {
```

Comment:
It'd be good to see some more "bad" data tested here. What if one of the provided entities is null? What if a number is provided instead of a string? What if the display template is null, or a blank string? I don't know what happens in these circumstances right now, so we should make sure nothing bad happens by testing any likely failures.

Reply:
All of these tests (except for empty template) would be covered by flow type annotations. I'll add them anyway for the moment. What currently happens:

1) one of the entities is null - `TypeError: cannot read property 'defautAlias' of null`
2) number is provided instead of a string - it works in the same exact way
3) `displayTemplate` is null - Handlebars complains `Error: You must pass a string or Handlebars AST to Handlebars.compile. You passed null`
4) `displayTemplate is the empty string - it returns the empty string as well